packages/nuxt/src/pages/runtime/router.options.ts

```diff
@@ -65,3 +74,7 @@ function _getHashElementScrollMarginTop (selector: string): number {
   }
   return 0
 }
+
+function _isDifferentLayout (from: RouteLocationNormalized, to: RouteLocationNormalized): boolean {
+  return from.meta.layout !== to.meta.layout
```

Comment:
I think there might be one thing to improve here - there might be an edge case where there is a missing `<NuxtLayout>`. For example, if `<NuxtLayout>` is used _within_ a page.

Maybe:

```suggestion
  return from.meta.layout && from.meta.layout !== to.meta.layout
```